src/lib/Components/Bidding/Screens/__tests__/ConfirmFirstTimeBid-tests.tsx

```diff
@@ -30,6 +44,87 @@ it("shows the billing address that the user typed in the billing address form",
   expect(billingAddressRow.findByType(Serif16).props.children).toEqual("401 Broadway 25th floor New York NY")
 })
 
+describe("successful bid", () => {
+  beforeEach(() => {
+    stripe.createTokenWithCard.mockReturnValueOnce(stripeToken)
+    relay.commitMutation = jest
+      .fn()
+      .mockImplementationOnce((_, { onCompleted }) => onCompleted())
+      .mockImplementationOnce((_, { onCompleted }) => onCompleted(mockRequestResponses.placeingBid.bidAccepted))
+  })
+
+  it("commits two mutations, one for createCreditCard followed by createBidderPosition", () => {
+    mockphysics.mockReturnValueOnce(Promise.resolve(mockRequestResponses.pollingForBid.highestBid))
+
+    const component = renderer.create(<ConfirmFirstTimeBid {...initialProps} />)
+
+    // manually setting state to avoid dplicating tests for skipping UI interation, but practically better not to do this.
+    component.root.instance.setState({ billingAddress })
+    component.root.instance.setState({ creditCardToken: stripeToken })
+    component.root.findByType(Checkbox).instance.props.onPress()
+    component.root.findByType(Button).instance.props.onPress()
+
+    jest.runAllTicks()
+
+    expect(relay.commitMutation).toHaveBeenCalledWith(
+      any(Object),
+      objectContaining({
+        variables: {
+          input: {
+            token: "fake-token",
+          },
+        },
+      })
+    )
+
+    expect(relay.commitMutation).toHaveBeenCalledWith(
+      any(Object),
+      objectContaining({
+        variables: {
+          input: {
+            sale_id: saleArtwork.sale.id,
+            artwork_id: saleArtwork.artwork.id,
+            max_bid_amount_cents: 450000,
+          },
+        },
+      })
+    )
+  })
+
+  it("polls for new results", () => {
+    mockphysics.mockReturnValueOnce(Promise.resolve(mockRequestResponses.pollingForBid.pending))
+    mockphysics.mockReturnValueOnce(Promise.resolve(mockRequestResponses.pollingForBid.highestBid))
+
+    const component = renderer.create(<ConfirmFirstTimeBid {...initialProps} />)
+
+    component.root.instance.setState({ billingAddress })
+    component.root.instance.setState({ creditCardToken: stripeToken })
+    component.root.findByType(Checkbox).instance.props.onPress()
+    component.root.findByType(Button).instance.props.onPress()
+
+    jest.runOnlyPendingTimers()
+    jest.runAllTicks()
+    jest.runOnlyPendingTimers()
+    jest.runAllTicks()
```

Comment:
Why do we need this back and forth?

Reply:
Ahh simulating polling?

Reply:
Yes. We have to resolve promises for MP otherwise it wouldn't push a new component.